ffmpeg/ffmpeg_test.go

```diff
@@ -2424,3 +2425,111 @@ func TestTranscoder_PNGDemuxerOpts(t *testing.T) {
 	assert.Equal(t, 3, res.Decoded.Frames)
 	assert.Equal(t, 180, res.Encoded[0].Frames)
 }
+
+func TestTranscode_DurationLimit(t *testing.T) {
+	run, dir := setupTest(t)
+	defer os.RemoveAll(dir)
+	cmd := `
+		ffmpeg -f lavfi -i color=c=blue:s=1280x720 -r 1 -frames:v 301 -c:v libx264 test-dur-bad.ts
+		ffmpeg -f lavfi -i color=c=blue:s=1280x720 -r 1 -frames:v 300 -c:v libx264 test-dur-good.ts
```

Comment:
It is faster to use an existing sample and just modify it as needed, rather than trying to re-generate / re-encode new ones. Look at how the other test cases do it.

Eg:

```suggestion
		# generate a 1fps sample
		ffmpeg -i $1../transcoder/test.ts -c copy -bsf:v setts=ts=N/TB_OUT/1 -frames:v 301 -y test.ts
		# double check the sample actually has the characteristics we expect
		ffprobe -show_format test.ts  | grep duration=301.00
```

Also feels like we can probably skip the "good" tests? If we still want to exercise the boundary conditions, then encode to the smallest possible output file.